controllers/gateway_controller_reconciler_utils.go

```diff
@@ -186,3 +189,100 @@ func (r *GatewayReconciler) getGatewayConfigForGatewayClass(ctx context.Context,
 		Name:      gatewayClass.Spec.ParametersRef.Name,
 	}, gatewayConfig)
 }
+
+func (r *GatewayReconciler) ensureDataPlaneNetworkPolicy(
+	ctx context.Context,
+	gateway *gatewayDecorator,
+	dataplane *operatorv1alpha1.DataPlane,
+	controlplane *operatorv1alpha1.ControlPlane,
+) error {
+	networkPolicies, err := gatewayutils.ListNetworkPoliciesForGateway(ctx, r.Client, gateway.Gateway)
+	if err != nil {
+		if !k8serrors.IsNotFound(err) {
+			return err
+		}
+	}
+
+	numNetworkPolicies := len(networkPolicies)
+	if len(networkPolicies) > 1 {
```

Comment:
```suggestion
	if numNetworkPolicies > 1 {
```

Either use the variable that we just created or remove the it - the variable - and use `len(networkPolicies)`